### ISSUE TITLE
Possible staking issue for the Operations group

### ISSUE THREAD

**Issue by bwhm** [Joystream/joystream#2505]
TL;DR
An opening was created with both application and role stake. All unstaking parameters was set to `null`.

One applicant applied, but the opening was closed without hiring anyone. Both stakes appears to have been burned.

In a previous opening for the operations group, all unstaking parameters was set to null expect for exit and terminate role, role stake (both 100800 blocks). Four applicants, all four hired.

When hired, the application stake seems to have been burned. The role stake appears to be working as expected here.

### Opening 3/22
- Opening ID: 22
- WG Opening ID: 3
- Opening created at: [1137603](https://testnet.joystream.org/#/explorer/query/1137603)
```
________________ Opening details _________________

WG Opening ID                 3                                 
Opening ID                    22                                
Type                          Worker                            
Stage                         Complete                          
Last status change            ~ 2:56:00 AM 6/26/2021 (#1141735) 
Application stake             >= 100.0000 kJOY                  
Role stake                    >= 500.0000 kJOY                  

_______________ Unstaking periods ________________

Crowded Out Application Stake Unstaking Period Length:                  0 blocks 
Crowded Out Role Stake Unstaking Period Length:                         0 blocks 
Exit Role Application Stake Unstaking Period:                           0 blocks 
Exit Role Stake Unstaking Period:                                       0 blocks 
Fill Opening Failed Applicant Application Stake Unstaking Period:       0 blocks 
Fill Opening Failed Applicant Role Stake Unstaking Period:              0 blocks 
Fill Opening Successful Applicant Application Stake Unstaking Period:   0 blocks 
Review Period Expired Application Stake Unstaking Period Length:        0 blocks 
Review Period Expired Role Stake Unstaking Period Length:               0 blocks 
Terminate Application Stake Unstaking Period:                           0 blocks 
Terminate Role Stake Unstaking Period:                                  0 blocks 

________________ Applications (1) ________________

WG appl. ID     Appl. ID     Member        Stage         Appl. stake     Role stake     Total stake     
13              75           maxlevush     Unstaking     0               0              0        
```
Only applicant:
- Member Applied: [2130/maxlevush](https://testnet.joystream.org/#/members/maxlevush)
- Account ID: `5Gy9eib5P1vB4Tvr9u74J7rhxvHqv5qhTdP3xwFfkArHQWPe`
- Applied at: [1138553](https://testnet.joystream.org/#/explorer/query/1138553)
- Stake ID - role stake: 262 (500000tJOY)
- Stake ID - application stake: 263 (100000tJOY)

Staked IDs, before and after applying
```
# 262/263 at 1138552
{created: 0, staking_status: NotStaked}
{created: 0, staking_status: NotStaked}
# 262/263 at 1138553
{created: 1138553, staking_status: {"staked":{"staked_amount":500000,"staked_status":{"normal":null},"next_slash_id":0,"ongoing_slashes":{}}}}
{created: 1138553, staking_status: {"staked":{"staked_amount":100000,"staked_status":{"normal":null},"next_slash_id":0,"ongoing_slashes":{}}}}
```
Account Balance, before/after applying (5Gy9eib5P1vB4Tvr9u74J7rhxvHqv5qhTdP3xwFfkArHQWPe):
```
# at 1138552 
{nonce: 648, refcount: 2, data: {free: 2482343, reserved: 1650000, miscFrozen: 1057167, feeFrozen: 1057167}}
# at 1138553
{nonce: 649, refcount: 2, data: {free: 1882343, reserved: 1650000, miscFrozen: 1057167, feeFrozen: 1057167}}
```

- Opening filled at: [1141735](https://testnet.joystream.org/#/explorer/query/1141735)
- Only the one applicant, no applicant hired

StakedIds, before and after applying
```
# 262/263 at 1141734
{created: 1138553, staking_status: {"staked":{"staked_amount":500000,"staked_status":{"normal":null},"next_slash_id":0,"ongoing_slashes":{}}}}
{created: 1138553, staking_status: {"staked":{"staked_amount":100000,"staked_status":{"normal":null},"next_slash_id":0,"ongoing_slashes":{}}}}
# 262/263 at 1141735
{created: 1138553, staking_status: NotStaked}
{created: 1138553, staking_status: NotStaked}
```

Account Balance, before after opening filled:
```
{nonce: 651, refcount: 2, data: {free: 1907843, reserved: 1650000, miscFrozen: 1057167, feeFrozen: 1057167}}
{nonce: 651, refcount: 2, data: {free: 1907843, reserved: 1650000, miscFrozen: 1057167, feeFrozen: 1057167}}
```

Total Issuance, before after opening filled:
```
848970389
848370389
```

So the 600k appears to have been burned.

### Opening 2/16

- Opening ID: 16
- WG Opening ID: 2
- Opening created at: [1010572](https://testnet.joystream.org/#/explorer/query/1010572)
```
________________ Opening details _________________

WG Opening ID                 2                                 
Opening ID                    16                                
Type                          Worker                            
Stage                         Complete                          
Last status change            ~ 8:09:36 PM 6/25/2021 (#1137747) 
Application stake             >= 100.0000 kJOY                  
Role stake                    >= 500.0000 kJOY                  

_______________ Unstaking periods ________________

Crowded Out Application Stake Unstaking Period Length:                  0 blocks       
Crowded Out Role Stake Unstaking Period Length:                         0 blocks       
Exit Role Application Stake Unstaking Period:                           0 blocks       
Exit Role Stake Unstaking Period:                                       100,800 blocks 
Fill Opening Failed Applicant Application Stake Unstaking Period:       0 blocks       
Fill Opening Failed Applicant Role Stake Unstaking Period:              0 blocks       
Fill Opening Successful Applicant Application Stake Unstaking Period:   0 blocks       
Review Period Expired Application Stake Unstaking Period Length:        0 blocks       
Review Period Expired Role Stake Unstaking Period Length:               0 blocks       
Terminate Application Stake Unstaking Period:                           0 blocks       
Terminate Role Stake Unstaking Period:                                  100,800 blocks 

________________ Applications (4) ________________

WG appl. ID     Appl. ID     Member           Stage         Appl. stake     Role stake     Total stake     
9               62           freakstatic      Unstaking     0               500000         500000          
10              63           nexusfallout     Unstaking     0               500000         500000          
11              65           isonar           Unstaking     0               500000         500000          
12              73           stwoof           Unstaking     0               500000         500000  
```

Applicants:

- Members Applied: 306,4,2182,1578
AccountIds:
```
5FVxUFN6fQFXrrV4FvhEDQUwefzwu9g1SrZCryiKrFhUiVc1
5FMrFXCbhtdv4tG5XaPsG3MyS6u36ZfsdCoHZxk4cTGdE56D
5HT8Mty5U5hFKVwv24uFswX2e41vurU3F5rkLMoETR8XmjFy
5CoTKA1x4mox5CKjd4R7AC6sNazbeY5xXv7xDHv47A9jw2w7
```
- Applied at: `1010670,1018237,1031285,1135013`
- StakeId - role stake: 218,222,227,257 (500000tJOY)
- StakeId - application stake: 219,223,228,258 (100000tJOY)

- Opening filled at: [1137747](https://testnet.joystream.org/#/explorer/query/1137747)
All applicants hired

Total Issuance, before after opening filled:
```
848683397
848283397
```

Member 1578 exited at: 1146926
Unstaking (as expected)

Here, it appears 400k (all application stakes) got burned.

---

Did a single spotcheck for a storage opening - 7/18 - and everything seems to be working...
